internal/metrics/metrics.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/rs/zerolog/log"
+)
+
+type Metrics struct {
+	errorCounter prometheus.Counter
+}
+
+func NewMetrics(reg prometheus.Registerer) *Metrics {
+	m := &Metrics{
+		errorCounter: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: "crawler_errors_total",
+			Help: "Total number of errors encountered by the crawler",
+		}),
+	}
+
+	reg.MustRegister(m.errorCounter)
+
+	return m
+}
+
+func UpdateErrorCounter(m *Metrics) {
```

Comment:
Big IMO on this, but: our language for metric-related functions shouldn't be verbs, but nouns. Instead of `UpdateErrorCounter`, try `CrawlerError` or just `Error`